clar2wasm/tests/standard/unit_tests.rs

```diff
@@ -1826,3 +1826,297 @@ fn bit_not_int() {
+    assert_eq!(result[0].unwrap_i64(), 123);
+    assert_eq!(result[1].unwrap_i64(), 0);
+
+    // pow(3, 2) == 9
+    pow.call(
+        &mut store,
+        &[Val::I64(3), Val::I64(0), Val::I64(2), Val::I64(0)],
+        &mut result,
+    )
+    .expect("call to pow-int failed");
+    assert_eq!(result[0].unwrap_i64(), 9);
+    assert_eq!(result[1].unwrap_i64(), 0);
+
+    // pow(3, 3) == 27
+    pow.call(
+        &mut store,
+        &[Val::I64(3), Val::I64(0), Val::I64(3), Val::I64(0)],
+        &mut result,
+    )
+    .expect("call to pow-int failed");
+    assert_eq!(result[0].unwrap_i64(), 27);
+    assert_eq!(result[1].unwrap_i64(), 0);
+
+    // pow(3, 80) = large number
+    pow.call(
+        &mut store,
+        &[Val::I64(3), Val::I64(0), Val::I64(80), Val::I64(0)],
+        &mut result,
+    )
+    .expect("call to pow-int failed");
+    assert_eq!(result[0].unwrap_i64(), 4389419161382147137);
+    assert_eq!(result[1].unwrap_i64(), 8012732698178659004);
+
+    // pow(3, 81) overflows
+    pow.call(
+        &mut store,
+        &[Val::I64(3), Val::I64(0), Val::I64(81), Val::I64(0)],
+        &mut result,
+    )
+    .expect_err("expected overflow");
+
+    // pow(2, 126) = 1 << 126
+    pow.call(
+        &mut store,
+        &[Val::I64(2), Val::I64(0), Val::I64(126), Val::I64(0)],
+        &mut result,
+    )
+    .expect("call to pow-uint failed");
+    assert_eq!(result[0].unwrap_i64(), 0);
+    assert_eq!(result[1].unwrap_i64(), 0x4000000000000000u64 as i64);
+
+    // pow(2, 127) overflows
+    pow.call(
+        &mut store,
+        &[Val::I64(2), Val::I64(0), Val::I64(127), Val::I64(0)],
+        &mut result,
+    )
+    .expect_err("expected overflow");
+
+    // pow(-3, 2) = 9
+    pow.call(
+        &mut store,
+        &[Val::I64(-3), Val::I64(-1), Val::I64(2), Val::I64(0)],
+        &mut result,
+    )
+    .expect("call to pow-int failed");
+    assert_eq!(result[0].unwrap_i64(), 9);
+    assert_eq!(result[1].unwrap_i64(), 0);
+
+    // pow(-3, 3) = -27
+    pow.call(
+        &mut store,
+        &[Val::I64(-3), Val::I64(-1), Val::I64(3), Val::I64(0)],
+        &mut result,
+    )
+    .expect("call to pow-int failed");
+    assert_eq!(result[0].unwrap_i64(), -27);
+    assert_eq!(result[1].unwrap_i64(), -1);
+
+    // edge case i128::MIN^1 is ok
+    pow.call(
+        &mut store,
+        &[
+            Val::I64(0),
+            Val::I64(0x8000000000000000u64 as i64),
+            Val::I64(1),
+            Val::I64(0),
+        ],
+        &mut result,
+    )
+    .expect("call to pow-int failed");
+    assert_eq!(result[0].unwrap_i64(), 0);
+    assert_eq!(result[1].unwrap_i64(), 0x8000000000000000u64 as i64);
+
+    // edge case i128::MIN^2 overflows
+    pow.call(
+        &mut store,
+        &[
+            Val::I64(0),
+            Val::I64(0x8000000000000000u64 as i64),
+            Val::I64(2),
+            Val::I64(0),
+        ],
+        &mut result,
+    )
+    .expect_err("expected overflow");
+}
```

Comment:
For both signed and unsigned, can you add tests that does 0 ^ large number and 1 ^ large number, to be sure those don't get caught in some eager overflow check?

Reply:
I'll do you one better. We already have a property check for `0^something == 1` (or 0 if `something` is 0). I'll add another property test for `1 ^ something == 1`.

Reply:
Yeah, I guess that covers it, doesn't it? 😄